Maturity model.md (#315)
gruetter/innerSourceTest: patterns/2-structured/maturity-model.md

## Patlet

Teams have started adopting InnerSource. The practice is spreading to multiple

> departments. However, the understanding of what constitutes an InnerSource project
> is not widespread. The solution is to provide a maturity model to allow for teams to

Reviewer: ```suggestion
departments. However, the understanding of what constitutes an InnerSource project
varies. The solution is to provide a maturity model to allow for teams to
```

Based on the [feedback](https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/315/files#r661315766) from @MaineC I a modifying this sentence slightly.

The important piece here seems to be that different teams have different understandings of what InnerSource means. I also confirmed that this is the intended mean in the Problem statement, where it states: 
> ... the depth of understanding for the [InnerSource] concept will diverge ...

Given that, I think the change proposed here is easier to understand.

## Patlet

Teams have started adopting InnerSource. The practice is spreading to multiple
departments. Understanding of what constitutes an InnerSource project are wide

> spread though. The solution is to provide a maturity model to allow for teams to

Reviewer: I think the authors of this pattern actually meant something else here.

@MaineC @dicortazar could you help us.
With the phrase "wide spread", did you mean to say that different teams have a very different understanding of what InnerSource means?

Reviewer: Reading the patlet again, I think the authors might mean something entirely different actually.

My assumption is that they meant to say something like:
"There are many different definitions of what constitutes an InnerSource project"

If that is indeed true, then we would have to fix this accordingly.

Reviewer 2: We should definitely fix the grammar in the sentence above. I don't remember the original meaning - I think we are better off going for the original pull request and issue related to the pattern than try to rely on the leaky memory of individuals.

https://github.com/InnerSourceCommons/InnerSourcePatterns/pull/121 has a lot of discussion on the pattern.

However there's nothing in there related to this particular sentence. I think both, the old variant as well as the new one could be counted as valid:

If understanding of what constitutes an InnerSource project varies, one could also translate that as the understanding of what constitutes a correct and 100% compliant InnerSource project being not widespread.

The question we would have to answer then is if there's one 100% compliant way of running an InnerSource project...

Looking at the maturity model pattern, what using this pattern helps with is increasing the knowledge about areas touched by InnerSource in teams - essentially moving them beyond the "but we are doing pull requests so we are all running great InnerSource projects".

go through a self check and discover patterns and practices that they are not
yet aware of.

## Problem

When InnerSource adoption in an enterprise starts to increase, individual
mentoring of each project through one evangelist becomes unfeasible. Also, more
and more people are gaining at least a basic understanding of what it means to

> work in an InnerSource project. Looking at all InnerSource projects through the

Reviewer: ```suggestion
work in an InnerSource project. Looking at all InnerSource projects though the
```

The sentence only makes sense with "though". Changing it back.

* PP-0: Individuals and teams do not regularly disclose their plans or products to multiple stakeholders. No formal actions exists at the organization.
* PP-1: Individuals and teams give visibility to their plans or products to multiple stakeholders, before they are started. Shared roadmap.
* PP-2: There are already shared roadmaps with clear guidelines and contribution rules where stakeholders can provide feedback. However, this is not standardized across the organization and not all of the projects provide this info.
* PP-3: Roadmaps are shared by default and there is a standard process and homogeneous way to do this across the organization at the level of each InnerSource project. This contains clear rules to contribute and influence in the roadmap.

**Development Process & Tools**

InnerSource projects thrive when contributors become active and participate. As a result, making contribution easier should be balanced with pure technical goals.

> * DP-0: Each team follows its own development process and tools. They are not defined to share knowledge and artifacts outside development team. Siled development teams.

Reviewer: ```suggestion
* DP-0: Each team follows its own development process and tools. They are not defined to share knowledge and artifacts outside development team. Siloed development teams.
```

They did mean teams in silos here.

* DC-2: People feel like they know about—and are helping to shape—most (but not all) important decisions as those decisions are unfolding. Materials that are part of decision-making practices are available at defined project milestones.
* DC-3: People feel like they are a part of a shared, standard process for collective decision-making that the organization endorses. Materials that are part of decision-making practices are continuously accessible during work processes.

**Helpful Resources**

In order to attract contributors helpful supporting material needs to be easily accessible.

* RS-0: Individuals and teams neither contribute to nor draw upon a shared repository of knowledge.
* RS-1: Individuals and teams release project materials for review internally, after they've finished their work. Individuals and teams share resources, but in disconnected, fragmented, or individualized/siled systems or repositories. Individuals and teams share resources, but there is no commonly expressed or shared understanding of the criteria used to determine whether information is

> sensitive or not. Do not "share thinking on others".

Reviewer: ```suggestion
* RS-1: Individuals and teams release project materials for review internally, after they've finished their work. Individuals and teams share resources, but in disconnected, fragmented, or individualized/siloed systems or repositories. Individuals and teams share resources, but there is no commonly expressed or shared understanding of the criteria used to determine whether information is sensitive or not. Do not "share thinking on others".
```

Let's keep this in one row.

Also fixing "siloed" here.

For silos to be reduced colleagues need to be comfortable sharing feedback openly. One easy way to support that is to use the same communication principles across hierarchies.

> Ideally you will end up with proper communication channels that are known by anyone in the organization - with channels focussed on different goals (announcements, user support, development channels, infra discussions, etc.). Some of the best practices you will establish as your InnerSource projects mature: Adoption of netiquette guidelines, opening a proven set of standard channels (which are
> being archived, publicly accessible, searchable) for each new InnerSource project.

Reviewer: ```suggestion
Ideally you will end up with proper communication channels that are known by anyone in the organization - with channels focussed on different goals (announcements, user support, development channels, infra discussions, etc.). Some of the best practices you will establish as your InnerSource projects mature: Adoption of netiquette guidelines, opening a proven set of standard channels (which are being archived, publicly accessible, searchable) for each new InnerSource project.
```

Let's keep this in one row.

* CF-0: There are no processes nor established channels. Some members of the organization share materials via private channels or discussions.

> * CF-1: The organization is in the process of establishing internal guidelines and channels for encouraging diverse points of view about company/departmental decisions, so that anyone belonging to the organization can use them. Some members of the organization share decision-making materials informally using unofficial platforms. Leaders maintain at least one clear and direct channel for
> organization members to share opinions constructively on some matters relevant to their work.

Reviewer: ```suggestion
* CF-1: The organization is in the process of establishing internal guidelines and channels for encouraging diverse points of view about company/departmental decisions, so that anyone belonging to the organization can use them. Some members of the organization share decision-making materials informally using unofficial platforms. Leaders maintain at least one clear and direct channel for organization members to share opinions constructively on some matters relevant to their work.
```

Let's keep this in one row.

**InnerSource Roles**

> InnerSource comes with explicit roles. While in early stages some patterns may be usable without adopting those roles, communicating within projects using explicit role titles becomes easier.

Reviewer: ```suggestion
InnerSource comes with explicit roles. While in early stages some patterns may be useable without adopting those roles, communicating within projects using explicit role titles becomes easier.
```

As both spellings are correct, let's stick with the original here/

**InnerSource Roles**

InnerSource comes with explicit roles. While in early stages some patterns may be usable without adopting those roles, communicating within projects using explicit role titles becomes easier.

* RO-0: There are no specific roles helping InnerSource adoption. Only common development roles are present: developer, analyst, tester, etc.
* RO-1: Occasionally some individuals and teams contribute to other projects. These are technical contributions, where the user/contributor role is seen. For some teams, it can be identified at least one member being a technical reference, who explains the development process to other development team members. He/she could be a candidate for covering the trusted committer role.

> * RO-2: An InnerSource Officer role is in charge of governance and support, including processes, etc. Identifies the education needs and ensures it is provided to the organization. Leads and mentors the organization in the engagement in IS projects. Is the first formal step in the way, defining the IS vision and roadmap. The organization has defined a trusted committer role, being a point of
> contact/reference not only for dev team members but also for external contributors. There is a standard process describing how to contribute to the community, contributor role is present. Data Scientist role is in charge of managing the traces of activity left by the InnerSource initiative, needed to measure the IS evolution. Trusted committer role will evolve to a more technical profile, and a
> community manager will be in charge of "energizing" the community, being his main responsibility to attract and retain new developers/users (contributors/community members).

Reviewer: ```suggestion
* RO-2: An InnerSource Officer role is in charge of governance and support, including processes, etc. Identifies the education needs and ensures it is provided to the organization. Leads and mentors the organization in the engagement in IS projects. Is the first formal step in the way, defining the IS vision and roadmap. The organization has defined a trusted committer role, being a point of contact/reference not only for dev team members but also for external contributors. There is a standard process describing how to contribute to the community, contributor role is present. Data Scientist role is in charge of managing the traces of activity left by the InnerSource initiative, needed to measure the IS evolution. Trusted committer role will evolve to a more technical profile, and a community manager will be in charge of "energizing" the community, being his main responsibility to attract and retain new developers/users (contributors/community members).
```

Let's keep this in one row.
